x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -10,8 +10,7 @@ import (
 
 // TODO add unit tests for all the keepers
 
-// GetCurrentDataCommitment creates the latest data commitment at current height according to
-// the data commitment window specified.
+// GetCurrentDataCommitment returns a data commitment for the current height.
 // Assumes that this method is called when:
 //
 //	ctx.BlockHeight() != 0 && ctx.BlockHeight()%int64(k.GetDataCommitmentWindowParam(ctx)) == 0
```

Comment:
> in the body of this function, what assumes that the method is called when //	ctx.BlockHeight() != 0 && ctx.BlockHeight()%int64(k.GetDataCommitmentWindowParam(ctx)) == 0

The creation of the data commitment range. If you don't call it in that condition, it will give you a different data commitment from the one expected by end block.

I consider assume to mean a precondition on calling the function.

Reply:
> I consider assume to mean a precondition on calling the function.

Agreed. Instead of a precondition that may or may not hold true, can the function body enforce that this precondition is true or throw an error if false?

Reply:
ok let's resolve this thread. I don't think my suggestion is super helpful.